File: testes/Lucas/Problema1/PPH/gera_pph.py
import math
import random
import time

MaxDim = 100
Rand_Max = 100000

def seedRandom(seed):
    random.seed(seed)

def randomn():
    return random.random()

def seconds():
    return time.process_time()

def randshannon():
    global I_SEED
    I_SEED = 16807 * I_SEED

    if I_SEED < 0:
        I_SEED += 2147483647

    i = I_SEED // 32767
    j = I_SEED - i * 32767

    random = j / 32767.0

    return random

def gera_conj_int(n, n_inst):
    global I_SEED
    Dim = n
    vt = [0] * MaxDim

    I_SEED = 123456789

    print("n:", n)
    print("n_inst:", n_inst)

    cputime = seconds()

    for j in range(1, n_inst + 1):
        num = str(n)
        print("Num:", num)
        j1 = str(j // 10)
        j2 = str(j - (10 * int(j1)))
        j1 += '0'
        j2 += '0'

        file = "pph_" + num + "_" + j1 + j2 + ".dat"

        with open(file, "wt") as p_in:
            p_in.write(str(n) + "\n")

            cont = 0

            for i in range(1, n + 2):
                p_in.write(" {:.0f}".format(randshannon() * 500))
                cont += 1
                if cont == 10:
                    cont = 0
                    p_in.write("\n")

            cont = 1
            x = int(randshannon() * 1000)
            if x == 0:
                x = 117
            p_in.write("\n {}".format(x))

            for i in range(1, n + 1):
                p_in.write(" {:.0f}".format(randshannon() * 1000))
                cont += 1
                if cont == 10:
                    cont = 0
                    p_in.write("\n")

    print(" CPU Time:", seconds() - cputime)

if __name__ == "__main__":
    Dim = 0
    vt = [0] * MaxDim
    I_SEED = 123456789

    Dim = int(input("n: "))
    v = int(input("n_inst: "))
    cputime = seconds()
    gera_conj_int(Dim, v)
    print(" CPU Time:", seconds() - cputime)

File: testes/Lucas/Problema1/PPH/gera_pph.c

#include <math.h>
#include <stdlib.h>
#include <stdio.h>
//#include <conio.h>
#include <string.h>
#include <sys/types.h>

// #include <iostream.h>
// #include <iomanip.h>
#include <time.h>     // for seconds()

#define MaxDim  100
#define Rand_Max  100000

// Prototypes

void seedRandom(unsigned int );     
double randomn(void );
double seconds(void );
double randshannon ( void );

void gera_conj_int(int, int );


long int I_SEED = 123456789;

int Dim;
int vt[MaxDim];

// Programa Principal

int main()
{
	int v;

	double cputime;

	printf("n: ");
	scanf("%d", &Dim);

	printf("n_inst: ");
	scanf("%d", &v);

	cputime = seconds();
    gera_conj_int(Dim, v);
    printf(" CPU Time: %lf\n", seconds() - cputime);

	return(0);
}


//  Gera Conjunto de Inteiros
//
//

void gera_conj_int(int n, int n_inst)
{
	int i,j,cont,x;
    FILE  *p_in;
    char  file[13], j1[2], j2[2], num[8];


	itoa(n, num, 10);
	printf("Num: %s\n",num);

	for(j=1; j<= n_inst; j++)
	{
      j1[0] = j / 10;
      j2[0] = j - ( 10 * j1[0] );

      j1[0] +=48;
      j2[0] +=48;
      j1[1] = '\0';
      j2[1] = '\0';


	  strcpy(file, "pph_");

      strcat(file,num);
      strcat(file,"_");
      strcat(file,j1);
      strcat(file, j2);
      strcat(file,".dat");

	  p_in = fopen(file, "wt");
	  fprintf(p_in,"%d\n", n);
      

	  cont = 0;

	  for(i=1; i<=n+1; i++)
	  {
 	    fprintf(p_in," %d",(int)(randshannon() * 500.));
		cont++;
		if(cont == 10)
		{
			cont = 0;
		    fprintf(p_in, "\n");
		}
	  }

      cont = 1;
      x = (int)(randshannon() * 1000.);
      if( x == 0 )
      {
          x = 117;
      }
	  fprintf(p_in, "\n %d",x);

	  for(i=1; i<=n; i++)
	  {
 	    fprintf(p_in," %d",(int)(randshannon() * 1000.));
		cont++;
		if(cont == 10)
		{
			cont = 0;
		    fprintf(p_in, "\n");
		}
	  }

	  fclose(p_in);
	}
}

 
double seconds()
// cpu time in seconds since start of run.
{
  double secs;
   
  secs = (double)(clock() / 1000.0);
  return(secs);
}


// random number between 0.0 and 1.0  Shannon random number generator.
double randshannon ( void )
{
  double   random;
  long int i,j;

  I_SEED = 16807 * I_SEED;

  if (I_SEED < 0)
  {
    I_SEED += 2147483647;
  }

  i = I_SEED / 32767;
  j = I_SEED - i * 32767;

  random = j/32767.;

  /*  printf("%f\n",*random); */

  return (random);
}




